pyomo/contrib/doe/fim_doe.py

```diff
@@ -49,32 +120,15 @@ def __init__(self, param_init, design_variable_timepoints, measurement_variable_
         self.design_name = list(self.design_timeset.keys())
         # the control time point for each design variable
         self.design_time = list(self.design_timeset.values())
-        # model output (measurement) name
-        self.measurement_variable_timepoints = measurement_variable_timepoints
-        self.measurement_variables = list(measurement_variable_timepoints.keys())
-        # model measurement time
-        self.measurement_timeset = list(measurement_variable_timepoints.values())
         # create_model()
         self.create_model = create_model
         self.args = args
-        self.measurement_extra_index = measurement_extra_index
-        extra_measure_name = list(measurement_extra_index.keys())
-        extra_measure_value = list(measurement_extra_index.values())
-        # check if measurement variables need to be flattened
-        if self.measurement_extra_index is not None:
-            flatten_measure_name=[]
-            for j in self.measurement_variables:
-                if j in extra_measure_name:
-                    for ind in measurement_extra_index[j]:
-                        flatten_name = j + str(ind)
-                        flatten_measure_name.append(flatten_name)
-                else:
-                    flatten_measure_name.append(j)
-        else:
-            flatten_measure_name = self.measurement_variables.copy()
 
-        self.flatten_measure_name = flatten_measure_name
+        # create the measurement information object
+        self.measure = Measurement_flatten(measurement_variable_timepoints, measurement_extra_index)
```

Comment:
Alternate design: the user provides a valid measurement object. Information such as `flatten_measurement_name` and `flatten_measurement_timeset` should be stored in the measurement object.